github-runner-manager/tests/unit/test_cli_config.py

```diff
@@ -0,0 +1,211 @@
+#  Copyright 2025 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Unit test for the cli_config module."""
+
+from pathlib import Path
+
+import pydantic
+import pytest
+import yaml
+
+from src.github_runner_manager.cli_config import Configuration
+
+ConfigValue = None | int | bool | str | tuple
+
+SAMPLE_CONFIG = {
+    "name": "test_org/test_repo",
+    "github_path": "test_org",
+    "github_token": "test_token",
+    "github_runner_group": None,
+    "runner_count": 1,
+    "runner_labels": ("test", "unit-test", "test-data"),
+    "openstack_auth_url": "http://www.example.com/test_url",
+    "openstack_project_name": "test-project",
+    "openstack_username": "test-username",
+    "openstack_password": "test-password",
+    "openstack_user_domain_name": "default",
+    "openstack_domain_name": "default",
+    "openstack_flavor": "test_flavor",
+    "openstack_network": "test_network",
+    "dockerhub_mirror": None,
+    "repo_policy_compliance_url": None,
+    "repo_policy_compliance_token": None,
+    "enable_aproxy": True,
+}
+
+
+def get_cli_configuration(config: dict[str, ConfigValue], file_path: Path) -> Configuration:
+    """Get the Configuration object from given configuration dict.
+
+    Args:
+        config: A dict containing the configurations.
+        file_path: The path to a temp file. The yaml module only support dump to stream, might as
+            well use a temp file.
+
+    Returns:
+        The Configuration object created from the data in the dict.
+    """
+    with open(file_path, mode="w", encoding="utf-8") as file:
+        yaml.safe_dump(config, file)
+
+    with open(file_path, mode="r", encoding="utf-8") as file:
+        return Configuration.from_yaml_file(file)
+
+
+@pytest.fixture(name="tmp_yaml_file", scope="function")
+def tmp_yaml_file_fixture(tmp_path: Path) -> Path:
+    return tmp_path / "tmp_config.yaml"
+
+
+def test_sample_config(tmp_yaml_file: Path):
+    """
+    arrange: None.
+    act: Create the Configuration from SAMPLE_CONFIG.
+    assert: The configuration content should match.
+    """
+    config = get_cli_configuration(SAMPLE_CONFIG, tmp_yaml_file)
+    diff = set(config.dict().items()) ^ set(SAMPLE_CONFIG.items())
+    assert not diff
+
+
+def test_optional_config(tmp_yaml_file: Path):
+    """
+    arrange: Remove the optional entry in the SAMPLE_CONFIG.
+    act: Create the configuration from config.
+    assert: The configuration content should match.
+    """
+    config = dict(SAMPLE_CONFIG)
```

Comment:
as mentioned earlier, if we have a fixture we wouldn't need a dict call here

Reply:
There are some places the dict still needs to be copied. I have update to using a fixture for `SAMPLE_CONFIG` and remove the dict copy where it is not needed.